### ISSUE TITLE
IPA v.1.5 not working despite attempts to de-bug

### ISSUE THREAD

**Issue by kcl58759** [PacificBiosciences/pbbioconda#448]
We are trying to run IPA v.1.5. and have been unable to successfully run pbipa on UGA Sapelo2 cluster despite trouble shooting.
No matter what we get the error message:

The following have been reloaded with a version change:
1) GCCcore/10.2.0 => GCCcore/8.3.0
2) XZ/5.2.5-GCCcore-10.2.0 => XZ/5.2.4-GCCcore-8.3.0
3) binutils/2.35-GCCcore-10.2.0 => binutils/2.32-GCCcore-8.3.0
4) bzip2/1.0.8-GCCcore-10.2.0 => bzip2/1.0.8-GCCcore-8.3.0
5) ncurses/6.2-GCCcore-10.2.0 => ncurses/6.1-GCCcore-8.3.0
6) zlib/1.2.11-GCCcore-10.2.0 => zlib/1.2.11-GCCcore-8.3.0

PackagesNotFoundError: The following packages are missing from the target environment:
- python=3.1

PackagesNotFoundError: The following packages are not available from current channels:

- python=3.1

Current channels:

- https://conda.anaconda.org/conda-forge/linux-64
- https://conda.anaconda.org/conda-forge/noarch
- https://conda.anaconda.org/bioconda/linux-64
- https://conda.anaconda.org/bioconda/noarch
- https://repo.anaconda.com/pkgs/main/linux-64
- https://repo.anaconda.com/pkgs/main/noarch
- https://repo.anaconda.com/pkgs/r/linux-64
- https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

https://anaconda.org

and use the search bar at the top of the page.

INFO: /apps/eb/pbipa/1.5.0/bin/ipa dist -i 1033.css.fq --nthreads 20 --njobs 1 --run-dir 1033.ipa --cluster-args 'qsub -S /bin/bash -N ipa -cwd -q default -pe smp {params.num_threads} -e qsub_log/ -o qsub_log/ -V'
INFO: ipa.py ipa (wrapper) version=1.5.0 ... Checking dependencies ...
samtools: error while loading shared libraries: libcrypto.so.1.0.0: cannot open shared object file: No such file or directory
INFO: Dependencies
/apps/eb/pbipa/1.5.0/bin/python3
/apps/eb/pbipa/1.5.0/bin/ipa2-task
/apps/eb/pbipa/1.5.0/bin/falconc
/apps/eb/pbipa/1.5.0/bin/minimap2
/apps/eb/pbipa/1.5.0/bin/nighthawk
/apps/eb/pbipa/1.5.0/bin/pancake
/apps/eb/pbipa/1.5.0/bin/pblayout
/apps/eb/pbipa/1.5.0/bin/racon
/apps/eb/pbipa/1.5.0/bin/samtools
/apps/eb/pbipa/1.5.0/bin/ipa_purge_dups
/apps/eb/pbipa/1.5.0/bin/ipa_purge_dups_split_fa
snakemake version=6.10.0
ipa2-task 1.5.0 (commit c875fce13bdacbafc2f4f750c6438f4453e1354d)
Machine name: 'Linux'
Copyright (C) 2004-2021 Pacific Biosciences of California, Inc.
This program comes with ABSOLUTELY NO WARRANTY; it is intended for
Research Use Only and not for use in diagnostic procedures.

We have tried:

install pbipa as recommended - openssl is the wrong version (libcrypto.so.1.0.0 error)
conda install pbipa openssl=1.0 -c defaults -c conda-forge -c bioconda - doesn't install all of the scripts needed (e.g. ipa_purge_dups_split_fa)
conda install pbipa samtools=1.9 -c defaults -c conda-forge -c bioconda - same as above
install pbipa as recommended, then try to downgrade openssl to 1.0 in the environment after installation - doesn't work, package conflicts
install pbipa as recommended, then try to upgrade samtools to 1.9 in the environment after installation - same as above
install pbipa as recommended, then conda update samtools - does nothing
install pbipa as recommended, then conda update -c conda-forge -c bioconda -c defaults samtools - does nothing

**Comment by isovic** [comment]
Hi @kcl58759 ,

This is a known long-standing Samtools error which happens with older versions of Samtools (v1.9).
I tried reproducing this issue multiple times, but in all cases Samtools `1.15` gets installed, and everything works fine.

Most likely, Samtools or OpenSSL get pulled from the wrong Conda channel.
The same issue was also reported in other projects with some suggested solutions:
https://github.com/bioconda/bioconda-recipes/issues/12100

All the reports we have seen were with respect to Samtools v1.9.
**The best solution is to upgrade Samtools to >=1.10.**

Otherwise, the solution which worked for most users is to manually create a file `~/.condarc` with the following content:
```
channels:
  - conda-forge
  - bioconda
  - defaults
```
If the order of channels is wrong, Conda may pull wrong versions of Samtools and libcrypto.

Another solution is to explicitly downgrade `openssl`, and I saw you already tried something similar to this but with a different set of `conda` channels listed:
```
conda install -c bioconda samtools openssl=1.0
```

Make sure you install `pbipa` fresh after these changes (e.g. in a new environment).

Does any of this fix the issue for you?

Best regards,
Ivan.
